library/BrazeEngine.m

```diff
@@ -0,0 +1,129 @@
+/*
+ * Use of this software is subject to the terms and
+ * conditions of the license agreement between you
+ * and Factual Inc
+ *
+ * Copyright © 2019 Factual Inc. All rights reserved.
+ */
+
+#import "Appboy_iOS_SDK/AppboyKit.h"
+#import "BrazeEngine.h"
+#import "PlaceCategoryMap.h"
+#import "PlaceChainMap.h"
+
+@implementation BrazeEngine
+
+NSString *SOURCE_NAME = @"factual";
+
+NSString *CIRCUMSTANCE_MET_EVENT_KEY = @"engine_";
+NSString *INCIDENT_ID_KEY = @"incident_id";
+NSString *USER_LATITUDE_KEY = @"user_latitude";
+NSString *USER_LONGITUDE_KEY = @"user_longitude";
+NSString *EVENT_SOURCE_KEY = @"event_source";
+
+NSString *AT_PLACE_EVENT_KEY = @"engine_at_";
+NSString *NEAR_PLACE_EVENT_KEY = @"engine_near_";
+NSString *PLACE_NAME_KEY = @"name";
+NSString *PLACE_ID_KEY = @"factual_id";
+NSString *PLACE_LATITUDE_KEY = @"latitude";
+NSString *PLACE_LONGITUDE_KEY = @"longitude";
+NSString *PLACE_CATEGORIES_KEY = @"category_labels";
+NSString *PLACE_CHAIN_KEY = @"chain_name";
+
+static BOOL trackingSpans = false;
```

Comment:
Does `false` work without error for `BOOL`? We tend to use `true`/`false` in our Obj-C++ code, since C++ defines them, but otherwise I thought Obj-C used the macros `YES` and `NO`.